btree/btree.c

```diff
@@ -8229,132 +8229,69 @@ static int64_t btree_truncate_tick(struct m0_sm_op *smop)
 			return m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_FINI);
 		}
 	case P_ACT: {
-		uint64_t        count = oi->i_used;
-		int             i;
-		int             rec_count;
-		struct slot     node_slot = {};
-		struct segaddr  child;
-		struct level   *parent;
+		int           rec_count;
+		struct slot   node_slot = {};
+		struct level *parent;
 
-		lev = &oi->i_level[count];
+		lev = &oi->i_level[oi->i_used];
 
+		/**
+		 * lev->l_node points to the node that needs to be truncated.
+		 * Set record count as zero and handle root/non-root case.
+		 */
+		bnode_set_rec_count(lev->l_node, 0);
+		node_slot.s_node = lev->l_node;
+		node_slot.s_idx  = 0;
 		/**
 		 * Case: root node as leaf node.
-		 * Set record count as 0, capture in transaction and exit state
+		 * Mark it as leaf node, capture in transaction and exit state
 		 * machine.
 		 */
-		if (count == 0) {
+		if (oi->i_used == 0) {
 			bnode_set_level(lev->l_node, 0);
-			bnode_set_rec_count(lev->l_node, 0);
-			node_slot.s_node = lev->l_node;
-			node_slot.s_idx  = 0;
 			bnode_capture(&node_slot, bop->bo_tx);
 			lock_op_unlock(tree);
 			return m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_FINI);
 		}
 		/**
 		 * Case: Tree has more than one level.
-		 * Tree traversal hase reached the leaf node.Take parent node of
-		 * leaf and delete all its child nodes.
-		 * Before deleting all the nodes in parent, delete the last
-		 * accessed node as we do not want to call node_get again for
-		 * this node.
+		 * Tree traversal has reached the leaf node. Delete the leaf
+		 * node.
 		 */
+		bnode_capture(&node_slot, bop->bo_tx);
```

Comment:
Call bnode_capture() after calling bnode_fini() on the next line so that the header magic change is also captured.

Reply:
Oh yes. Thanks.